dcpy/lifecycle/ingest/run.py

```diff
@@ -49,34 +50,46 @@ def run(
     latest: bool = False,
     skip_archival: bool = False,
     output_csv: bool = False,
+    template_dir: Path = configure.TEMPLATE_DIR,
 ) -> Config:
-    config = configure.get_config(dataset_id, version=version, mode=mode)
-    transform.validate_processing_steps(config.id, config.processing_steps)
+    config = configure.get_config(
+        dataset_id, version=version, mode=mode, template_dir=template_dir
+    )
+    transform.validate_processing_steps(config.id, config.ingestion.processing_steps)
```

Comment:
is this redundant since `transform.validate_process_steps` is called during `transform.preprocess`?

and if we do wanna validate early, would it make sense to do it during `configure.get_config`?

Reply:
talked about it and we'll keep it here. it's nicer to assemble these pieces at a high-level rather than have ingest modules importing each other